src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -164,21 +165,21 @@ object Desugar {
 
     private val nm = new NameManager
 
-    type Identity = Meta
+    type Identity = (Meta, TypeInfo)
 
-    private def abstraction(id: PIdnNode): Identity = {
+    private def abstraction(id: PIdnNode, info: TypeInfo = info): Identity = {
       val entity = info.regular(id)
-      meta(entity.rep)
+      (meta(entity.rep, entity.context.getTypeInfo), entity.context.getTypeInfo)
     }
```

Comment:
why do we now need to pass `info` as a parameter?

Reply:
It's to avoid getting a `node not in tree exception` (as you said in the other comment)